extensions/blocks/pinterest/edit.js

```diff
@@ -0,0 +1,129 @@
+import { Placeholder, SandBox, Button, IconButton, Toolbar } from '@wordpress/components';
+import { BlockControls } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { pinType } from './utils';
+import { icon } from '.';
+import './editor.scss';
+
+class PinterestEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.hideOverlay = this.hideOverlay.bind( this );
+		this.setUrl = this.setUrl.bind( this );
+
+		this.state = {
+			editedUrl: this.props.attributes.url,
+			editingUrl: false,
+			// The interactive-related magic comes from Core's EmbedPreview component,
+			// which currently isn't exported in a way we can use.
+			interactive: false,
+		};
+	}
+
+	static getDerivedStateFromProps( nextProps, state ) {
+		if ( ! nextProps.isSelected && state.interactive ) {
+			// We only want to change this when the block is not selected, because changing it when
+			// the block becomes selected makes the overlap disappear too early. Hiding the overlay
+			// happens on mouseup when the overlay is clicked.
+			return { interactive: false };
+		}
+
+		return null;
+	}
+
+	hideOverlay() {
+		// This is called onMouseUp on the overlay. We can't respond to the `isSelected` prop
+		// changing, because that happens on mouse down, and the overlay immediately disappears,
+		// and the mouse event can end up in the preview content. We can't use onClick on
+		// the overlay to hide it either, because then the editor misses the mouseup event, and
+		// thinks we're multi-selecting blocks.
+		this.setState( { interactive: true } );
+	}
+
+	setUrl( event ) {
+		if ( event ) {
+			event.preventDefault();
+		}
+
+		const { editedUrl: url } = this.state;
+		this.setState( { editingUrl: false } );
+		this.props.setAttributes( { url } );
+	}
+
+	/**
+	 * Render a preview of the Pinterest embed.
+	 *
+	 * @returns {object} The UI displayed when user edits this block.
+	 */
+	render() {
+		const { attributes, className } = this.props;
+		const { url } = attributes;
+		const { editedUrl, interactive, editingUrl } = this.state;
+
+		const showEditor = editingUrl || ! url;
+		const type = pinType( url );
+		const html = `<a data-pin-do='${ type }' href='${ url }'></a>`;
+
+		// Disabled because the overlay div doesn't actually have a role or functionality
+		// as far as the user is concerned. We're just catching the first click so that
+		// the block can be selected without interacting with the embed preview that the overlay covers.
+		/* eslint-disable jsx-a11y/no-static-element-interactions */
+		return (
+			<div className={ className }>
+				<BlockControls>
+					<Toolbar>
+						<IconButton
+							className="components-toolbar__control"
+							label={ __( 'Edit URL', 'jetpack' ) }
+							icon="edit"
+							onClick={ () => this.setState( { editingUrl: true } ) }
+						/>
+					</Toolbar>
+				</BlockControls>
+				{ showEditor && (
+					<Placeholder label={ __( 'Pinterest', 'jetpack' ) } icon={ icon }>
```

Comment:
I think if you use `BlockIcon` component (`icon={ <BlockIcon icon={ icon } /> }`), you don't have to  add the CSS to set the size to 24px.

Reply:
👍🏻 06a4fbe78